File: python-01/ex2/ft_plant_growth.py
#!/usr/bin/env python3


class Plant:
    def __init__(self, name, height, days):
        self.name = name
        self.height = height
        self.days = days

    def grow(self, cm):
        self.height += cm

    def age(self, duration):
        self.days += duration

    def get_info(self):
        print(f"{self.name}: {self.height}cm, {self.days} days old")


if __name__ == "__main__":
    rose = Plant("Rose", 25, 30)
    print("=== Day 1 ===")
    rose.get_info()
    first_day_height = rose.height
    print("=== Day 7 ===")
    rose.grow(6)
    rose.age(6)
    rose.get_info()
    print(f"Growth this week: +{rose.height - first_day_height}cm")

File: cpp-01/ex2/ft_plant_growth.cpp
#include <iostream>
#include <string>
#include "Plant.hpp"

int main(int argc, char** argv)
{
    Plant rose("Rose", 25, 30);
    std::cout << "=== Day 1 ===" << std::endl;
    rose.get_info();
    std::cout << std::endl;
    int first_day_height = rose.height;
    std::cout << "=== Day 7 ===" << std::endl;
    rose.grow(6);
    rose.age(6);
    rose.get_info();
    std::cout << std::endl;
    std::cout << "Growth this week: +" << rose.height - first_day_height << "cm";
    std::cout << std::endl;
    return 0;
}
